pkg/identity/certificate_authority.go

```diff
@@ -178,16 +179,20 @@ func (pc PeerCAConfig) Load() (*PeerCertificateAuthority, error) {
 func NewCA(ctx context.Context, opts NewCAOptions) (
 	rv *FullCertificateAuthority, err error) {
 	defer mon.Task()(&ctx)(&err)
+	var (
+		highscore uint32
+	)
 
 	if opts.Concurrency < 1 {
 		opts.Concurrency = 1
 	}
 	ctx, cancel := context.WithCancel(ctx)
 
+	log.Printf("Generating a certificate matching a difficulty of %d\n", opts.Difficulty)
 	eC := make(chan error)
 	caC := make(chan FullCertificateAuthority, 1)
 	for i := 0; i < int(opts.Concurrency); i++ {
-		go newCAWorker(ctx, opts.Difficulty, opts.ParentCert, opts.ParentKey, caC, eC)
+		go newCAWorker(ctx, i, &highscore, opts.Difficulty, opts.ParentCert, opts.ParentKey, caC, eC)
 	}
 
 	select {
```

Comment:
Can we just use a for loop with delays and print `.`?

Reply:
That's not nearly as fun, and I've already figured out how to do it this way. ;)

Reply:
```
fmt.Print("generating certificate: ")
defer fmt.Println()
for {
	select {
	case <-ticker.C:
		fmt.Print(".")
	case ca := <-caC:
		return &ca, nil
	case err := <-eC:
		return nil, err
	case <-ctx.Done():
		return nil, ctx.Err()
	}
}
```